src/frameworks/us_privacy_string_and_api_generator.js

```diff
@@ -0,0 +1,147 @@
+const FrameworkBase = require('./base');
+
+const US_PRIVACY_API_VERSION = 1;
+
+/**
+ * Implements usprivacy string framework
+ * for more information on the US Privacy string see:
+ * https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/US%20Privacy%20String.md#us-privacy-string-format
+ */
+class UsPrivacyStringAndAPIGenerator extends FrameworkBase {
+  constructor() {
+    super();
+    this.window = window;
+    this.document = document;
+  }
+  supportedGenerators() {
+    return ['usPrivacyString', 'installPrivacyAPI'];
+  }
+
+  useConfig({ window, document }) {
+    this.window = window;
+    this.document = document;
+  }
+
+  usPrivacyString(callback = () => {}) {
+    callback(this.buildUsPrivacyString(), this);
+  }
+
+  installPrivacyAPI(callback = () => {}) {
+    this.window.__uspapi = this.handleUSPrivacyAPI.bind(this);
+    this.createUspapiFrame();
+    callback(true);
+  }
+
+  // Private methods ---------
+  buildUsPrivacyString() {
```

Comment:
_Minor/pedantry_: Could we standardise on capitalisation for "US Privacy String" in methods/variables? (I prefer `USPrivacy`, as below, but don't really care, as long as it's consistent.